src/jenesis/contracts/deploy.py

```diff
@@ -154,72 +164,105 @@ def deploy_contracts(cfg: Config, project_path: str, deployer_key: Optional[str]
 
     contracts = detect_contracts(project_path)
 
-    contracts_to_deploy = []  # type: List[Tuple[Contract, Deployment]]
     profile_contracts = selected_profile.contracts
     profile_contract_names = list(profile_contracts.keys())
 
-    # determine what tasks to do
-    for contract in contracts:
-        if contract.name in profile_contract_names:
-            profile_contract = selected_profile.deployments.get(contract.name)
-        else:
-            continue
-        assert profile_contract is not None
+    init_addresses = {contract.name: 
+                     {val for val in profile_contracts[contract.name]["init_addresses"]}
+                      for contract in contracts}
 
-        # ensure that contract has been compiled first
-        if not os.path.isfile(contract.binary_path):
-            print(f"No contract binary found for {contract.name}. Please run 'jenesis compile' first.")
-            continue
+    ts = gl.TopologicalSorter(init_addresses)
+    deployment_order = list(ts.static_order())
+    contract_to_deploy = ""
 
-        if deployer_key is not None:
-            profile_contract.deployer_key = deployer_key
-            Config.update_key(os.getcwd(), profile, contract, deployer_key)
+    for C in deployment_order:
 
-        # simple case the contract is already deployed and we can just use the information directly from the lockfile
-        if profile_contract.is_configuration_out_of_date():
-            contracts_to_deploy.append((contract, profile_contract))
-            continue
+        # determine what tasks to do
+        for contract in contracts:
 
-        digest = contract.digest()
-        if digest is None:
-            continue  # we can't process any contracts where we don't have
+            if contract.name != C:
+                continue
```

Comment:
This seems somewhat inefficient. Maybe better to set `contracts = {contract.name: contract for contract in detect_contracts(...)}` for easier indexing by name?

Reply:
Yes, you are right. Ill take a look